src/game_initialization/create_engine.cpp

```diff
@@ -791,11 +793,15 @@ void create_engine::init_extras(const MP_EXTRA extra_type)
 				extras.push_back(std::move(new_extras_metadata));
 			}
 			else {
-				//TODO: use a more visible error message.
 				ERR_CF << "found " << extra_name << " with id=" << extra["id"] << " twice";
+				duplicates.emplace_back(extra["id"].str());
 			}
 		}
 	}
+
+	if(duplicates.size() > 0) {
+		gui2::show_error_message("Found multiple modifications with the same id:\n"+utils::join(duplicates, "\n"));
```

Comment:
I think this should be translatable, even in 1.18.

There's two ways of looking at the string freeze, either measure it by what's in the PO files, or by the situations that cause English text to be shown to users. IMO the situation is serious enough to justify showing the string to the user, but if we're showing the string to the user anyway then there's no reason not to make it translatable.

Reply:
It is serious issue without this PR, but just debug info afterwards.

Reply:
That's good, but then it doesn't justify a GUI error dialog.

Reply:
Removed then.

Reply:
How does this PR accomplish making duplicate IDs not an error?

Reply:
I understand that after this change, when toggling duplicate option then all options with that id will be toggled, instead of random modification with different id.

Reply:
And how is that not an error?

Reply:
Maybe is maybe not, needs testing when next version is uploaded.

Reply:
In that case I think an error dialog is the right way to go, and it should also identify which add-ons are providing those modifications.

Reply:
It would enable whichever modification with that id it happens to find first. The thing fixed is that it doesn't make the UI behave in weird ways anymore.

Reply:
That sounds like a recipe for OOS :(

Reply:
The only way to possibly avoid the OOS risk would be to not display any of the modifications with duplicate IDs, making them unselectable and unusable.